runtime/compiler/trj9/runtime/J9CodeCacheManager.cpp

```diff
@@ -344,23 +344,29 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
                                                                       segmentType,
                                                                       &vmemParams);
 
+   uintptr_t someJitLibraryAddress = getSomeJitLibraryAddress();
+   // first isInRange() checks whether the allocated segment startAddress is in range to avoid trampoline
+   // second isInRange() checks if the preferredStartAddress is in range to avoid trampoline in the first place, for large pages (>1GB) it's possible they won't be in range
+   // if they are not, we will accept what we get 
    if (codeCacheSegment &&
        (vmemParams.options & J9PORT_VMEM_ADDRESS_HINT) &&
-       !isInRangeToAvoidTrampoline((uintptr_t)(codeCacheSegment->baseAddress)))
+       !isInRange((uintptr_t)(codeCacheSegment->baseAddress), someJitLibraryAddress, MAX_DISTANCE_NEAR_JITLIBRARY_TO_AVOID_TRAMPOLINE) &&
+       isInRange((uintptr_t)(preferredStartAddress), someJitLibraryAddress, MAX_DISTANCE_NEAR_JITLIBRARY_TO_AVOID_TRAMPOLINE))
```

Comment:
If I remember correctly the ADDRESS_HINT is ignored for large pages because the allocation mechanism is not based on mmap, but on shmget. If that's the case I would simplify the code by not setting `vmemParams.options |= J9PORT_VMEM_ADDRESS_HINT;` In this case and then eliminating this second `isInRange`